fdb-extensions/src/main/java/com/apple/foundationdb/ErrorCodes.java

```diff
@@ -0,0 +1,181 @@
+/*
+ * ErrorCodes.java
+ *
+ * This source file is part of the FoundationDB open source project
+ *
+ * Copyright 2015-2020 Apple Inc. and the FoundationDB project authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.apple.foundationdb;
+
+import com.apple.foundationdb.annotation.API;
+import com.google.common.collect.ImmutableMap;
+
+import javax.annotation.Nonnull;
+import java.util.Map;
+
+/**
+ * FDB error codes (from https://apple.github.io/foundationdb/api-error-codes.html) and handy methods to
+ * interpret them.
+ */
+@API(API.Status.MAINTAINED)
+public class ErrorCodes {
```

Comment:
Should this be an enum?

Reply:
@alecgrieser Sorry this has been stagnating so long.  I updated this to an `enum`.  It does have the benefit that we can later add other attributes to the error, but it does make other things a little more awkward - for example, there is no good way to represent an error code that we have never seen before.  But, I'm not unhappy with the change.

Reply:
I think I like the enum better, though it kind of seems like this should really be an enum that lives in the Java bindings, but alas. The `foundationdb` project already has some automation that can generate C++ files based on these error definitions, so it seems like they should be able to generate this `enum` as well.

There's an alternative version of this change that resolves #1020 but doesn't try and fix the problem of error codes (partially to avoid having two changes in one PR and partially to avoid trying to argue on how the API should look). I'm happy enough with how this is, though.